### PR TITLE
Feature/flex grid/store

### PR DESCRIPTION
**Pull request by notbakaneko**
Replacing bootstrap css from store-related pages with a flexbox-derived grid.

### CODE REVIEW

File: resources/assets/less/store.less [hunk 1]
```diff
@@ -18,5 +18,6 @@
 
 @import "app";
 
+@import "bem/flex-grid";
```
Comment (nanaya) [inline resources/assets/less/store.less]: `bem-index.less`.

File: resources/assets/less/bem/flex-grid.less [hunk 2]
```diff
@@ -0,0 +1,53 @@
+.grid {
```
Comment (nanaya) [inline resources/assets/less/bem/flex-grid.less]: license copypasta mising. Also one block per file, named with the block name.

File: resources/assets/less/bem/flex-grid.less [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    flex-wrap: wrap;
+  }
+
+  // Use this if we don't want the grid to collapse into a stack.
+  // FIXME: better name
+  &__no-block {
+    display: flex;
+  }
+
+  &--gutters {
```
Comment (nanaya) [inline resources/assets/less/bem/flex-grid.less]: modifiers should be declared before elements

File: resources/assets/less/bem/flex-grid.less [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  // size to let other cells expand and fill the gap.
+  &--squash {
+    flex: 0 1 0%;
+  }
+
+  &--fill {
+    flex: 1 0 100%;
+  }
+
+  // compiler doesn't like it if it starts with a number
```
Comment (nanaya) [inline resources/assets/less/bem/flex-grid.less]: hmmmm

File: resources/assets/less/bem/flex-grid.less [hunk 5]
```diff
@@ -0,0 +1,53 @@
+.grid {
+  // 840px seems a big large for this to trigger?
+  @media @desktop {
+    display: flex;
+    flex-wrap: wrap;
+  }
+
+  // Use this if we don't want the grid to collapse into a stack.
+  // FIXME: better name
+  &__no-block {
```
Comment (nanaya) [inline resources/assets/less/bem/flex-grid.less]: wait is this modifier or element D:

**Comment by notbakaneko** [comment]
Hang on, I think I trolled myself with the naming :trollface:

File: resources/assets/less/bem/grid.less [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  // Use this if we don't want the grid to collapse into a stack.
+  // FIXME: better name
+  &--xs {
+    display: flex;
+  }
+
+  &--gutters {
+    margin-left: -@flex-grid--gutter-width;
```
Comment (nanaya) [inline resources/assets/less/bem/grid.less]: shouldn't the gutters also apply to top and bottom? In case of wrapped cells.

Comment (nanaya) [inline resources/assets/less/bem/grid.less]: Or I guess one can just add margin bottom...

Comment (notbakaneko) [inline resources/assets/less/bem/grid.less]: Top and bottom spacing in grid column layouts tend to be different from the column gutters, so we don't really want to apply them together.

File: resources/assets/less/bem/grid-cell.less [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+.grid-cell {
+  flex: 1;
+  .grid--gutters > & {
+    padding-left: @flex-grid--gutter-width;
+    padding-right: @flex-grid--gutter-width;
+  }
+  // squashes the flex content down to the smallest possible
+  // size to let other cells expand and fill the gap.
+  &--squash {
```
Comment (nanaya) [inline resources/assets/less/bem/grid-cell.less]: can use sort by name

**Pull request merged**